src/devices/Ds1307/Ds1307.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Ds1307
+{
+    /// <summary>
+    /// Realtime Clock DS1307
+    /// </summary>
+    public class Ds1307 : IDisposable
+    {
+        /// <summary>
+        /// DS1307 Default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x68;
+
+        private I2cDevice _sensor = null;
+
+        /// <summary>
+        /// DS1307 DateTime
+        /// </summary>
+        public DateTime DateTime { get => ReadTime(); set => SetTime(value); }
+
+        /// <summary>
+        /// Creates a new instance of the DS1307
+        /// </summary>
+        /// <param name="sensor">I2C Device, like UnixI2cDevice or Windows10I2cDevice</param>
+        public Ds1307(I2cDevice sensor)
+        {
+            _sensor = sensor;
+        }
+
+        /// <summary>
+        /// Read Time from DS1307
+        /// </summary>
+        /// <returns>DS1307 Time</returns>
+        private DateTime ReadTime()
+        {
+            Span<byte> readBuffer = stackalloc byte[7];
+
+            _sensor.WriteByte((byte)Register.RTC_SEC_REG_ADDR);
```

Comment:
NIT: should we add a small comment here saying that you are reading all registers at the time instead of one by one?

Reply:
ok